### PR TITLE
samples: boards: pm on stm32 boards with serial wakeup

### LINKED ISSUE #49268: tests: samples/boards/stm32/power_mgmt/serial_wakeup failed on mec15xxevb_assy6853 (and several stm32 boards)
The daily twister tester will run samples/boards/stm32/power_mgmt/serial_wakeup on mec15xxevb_assy6853 and got a failure

Please also mention any information which could help others to understand
the problem you're facing:
 - What target platform are you using? -> mec15xxevb_assy6853
 
 - What have you tried to diagnose or workaround this issue?  
 Check the readme.rst in the test folder, and found this test case seems to be dedicated to stm32. Regarding to mec15xxevb_assy6853 has its own power_mgmt test case, this case should add "platform_alow: stm32" in the sample.yaml.

 - Is this a regression? If yes, have you been able to "git bisect" it to a
   specific commit?
 - Not a regression. It looks like a small bug in the commit f5378424f7f492a8ae0d173a015bc1186fcc7127.

**To Reproduce**
twister --hardware-map $HOME/mec1.file --device-testing -T samples/boards/stm32/power_mgmt/serial_wakeup/ -vv

**Expected behavior**
The test should be skipped

**Impact**
1 test case fail in daily test

**Logs and console output**
```
DEBUG   - DEVICE: *** Booting Zephyr OS build zephyr-v3.1.0-3848-g02b23f3733e0  ***
DEBUG   - DEVICE: Device ready
DEBUG   - DEVICE: Device is not wakeup capable
```

**Environment (please complete the following information):**
 - OS: (e.g. Linux, MacOS, Windows): Ubuntu 20.04
 - Toolchain (e.g Zephyr SDK, ...) 0.14.1
 - Commit SHA or Version used :02b23f3733e0e052775a7480ff1e7f6f7971f4e6

### PR DESCRIPTION
**Pull request by FRASTM**
This commit retrains the power management testcase
tests/samples/boards/stm32/power_mgmt/serial_wakeup
to bun on stm32 boards only.
Limiting first to nucleo_wg55 and stm32l562dk targets,
more to be added.

Fixes https://github.com/zephyrproject-rtos/zephyr/issues/49268

Signed-off-by: Francois Ramu <francois.ramu@st.com>

### CODE REVIEW

**Review by hunterhu3000 — DISMISSED** [review]

**Comment by hunterhu3000** [comment]
The patch looks good on mec15. The test case is skipped as expected
```
DEBUG   - platform filter: ['mec15xxevb_assy6853']
DEBUG   -     arch_filter: None
DEBUG   -      tag_filter: None
DEBUG   -     exclude_tag: None
INFO    - Building initial testsuite list...
DEBUG   - mec15xxevb_assy6853       samples/boards/stm32/power_mgmt/serial_wakeup/sample.boards.stm32.power_mgmt.serial_wakeup SKIPPED: Not in testsuite platform allow list
INFO    - Writing JSON report /home/ztest/zephyrproject/zephyr/twister-out/testplan.json
```

**Review by erwango — DISMISSED** [review]

File: samples/boards/stm32/power_mgmt/serial_wakeup/sample.yaml [hunk 1]
```diff
@@ -15,3 +15,4 @@ tests:
       - nucleo_wb55rg
     filter: dt_compat_enabled("zephyr,power-state")
     extra_args: "CONFIG_DEBUG=y"
+    platform_allow: nucleo_wb55rg stm32l562e_dk
```
Comment (gmarull) [inline samples/boards/stm32/power_mgmt/serial_wakeup/sample.yaml:18]: why don't you filter using stm32 uart compatible instead?

**Comment by erwango** [comment]
Please also remove check on `        - "Device ready"` in yaml file.
"Device ready" is no more printed since https://github.com/zephyrproject-rtos/zephyr/pull/49167

File: samples/boards/stm32/power_mgmt/serial_wakeup/sample.yaml (hunk 1 above)
Comment (erwango) [reply samples/boards/stm32/power_mgmt/serial_wakeup/sample.yaml:18]: Agreed.
This test will only work on the boards that are provided with an overlay to set up a clock configuration compatible with stop modes.

**Comment by FRASTM** [comment]
ADd a commit to "remove check on - "Device ready" in yaml file."

**Review by erwango — APPROVED** [review]

**Review by gmarull — APPROVED** [review]

**Pull request merged**